### PR TITLE
shipit_bot_uplift: add impacted branches to report email subject

### LINKED ISSUE #582: shipit_bot_uplift: the subject of the email should say which branches are impacted
Currently, the subject of the email is

> Uplift bot detected some merge failures

We should have

> Uplift bot detected some merge failures to beta, esr52

### PR DESCRIPTION
**Pull request by jankeromnes**
Fix #582.

### CODE REVIEW

**Comment by jankeromnes** [comment]
@La0 please have a look when you have some time.

**Comment by jankeromnes** [comment]
There are [no tests](https://github.com/mozilla-releng/services/blob/master/src/shipit_bot_uplift/tests/test_bot.py) for `shipit_bot_uplift`, so this change isn't covered.

I should add tests to this pull request.

**Comment by jankeromnes** [comment]
Attempted to add a test, but it fails with `Exception: Missing taskcluster in /etc/hosts`.

Apparently, this means that I need to somehow provide TaskCluster client credentials.

**Comment by jankeromnes** [comment]
Ok, I logged into https://tools.taskcluster.net/auth/clients/ with my Mozilla LDAP, and created a TaskCluster client with scope `secrets:get:repo:github.com/mozilla-releng/services:branch:staging`.

This allowed me to succesfully run my test locally:

```
$ sudo ./please check shipit-bot-uplift --taskcluster-client-id="$TC_ID" --taskcluster-access-token="$TC_TOKEN"
 => Checking code quality: DONE
 => Running tests: DONE
```

Slightly altering the expected email subject makes the test fail as expected:

```
E         - [test] Uplift bot detected 2 merge failures to beta,esr52
E         ?                            ^
E         + [test] Uplift bot detected 3 merge failures to beta,esr52
E         ?                            ^

tests/test_report.py:21: AssertionError
----------------------------- Captured stdout call -----------------------------
2018-04-16 15:28.10 Report notifications           emails=['test@mozilla.com']
====================== 1 failed, 1 passed in 0.70 seconds ======================
```

**Comment by jankeromnes** [comment]
Moving this workaround into a fixture doesn't seem to work.

File: src/shipit_bot_uplift/tests/test_report.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Any copyright is dedicated to the Public Domain.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+import json
+
+import responses
+
+
+@responses.activate
+def test_mail():
```
Comment (La0) [inline src/shipit_bot_uplift/tests/test_report.py]: Simply do:

```python
def test_mail(mock_taskcluster_credentials):
```

File: src/shipit_bot_uplift/tests/test_report.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    from shipit_bot_uplift.report import Report
+    from shipit_bot_uplift.merge import MergeTest
+
+    def _check_email(request):
+        payload = json.loads(request.body)
+
+        assert payload['subject'] == '[test] Uplift bot detected 2 merge failures to beta, esr52'
+        assert payload['address'] == 'test@mozilla.com'
+        assert payload['template'] == 'fullscreen'
+        assert payload['content'].startswith('# Failed automated merge test')
```
Comment (La0) [inline src/shipit_bot_uplift/tests/test_report.py]: :+1:

**Pull request merged**